src/hypothesis/internal/conjecture/engine.py

```diff
@@ -641,6 +732,11 @@ def exit_with(self, reason):
         self.exit_reason = reason
         raise RunIsComplete()
 
+    def begin_health_checks(self):
+        if not self.settings.perform_health_check:
```

Comment:
if self.settings.perform_health_check:
        self.health_check_state = HealthCheckState()

I think it might also be more readable as an inlined section with a brief explanatory comment - I would have had trouble following the effect of `self.health_check_state` if I hadn't been prepped by your writeup.